File: src/main/java/com/appointmed/dto/AuthResponse.java
package com.appointmed.dto;

import com.appointmed.model.User;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AuthResponse {
    private String token;
    private Long id;
    private String name;
    private String email;
    private String role;
    private String avatarUrl;

    public static AuthResponse from(User user, String token) {
        return AuthResponse.builder()
                .token(token)
                .id(user.getId())
                .name(user.getName())
                .email(user.getEmail())
                .role(user.getRole())
                .avatarUrl(user.getAvatarUrl())
                .build();
    }
}

File: mobile/app/src/main/java/com/appointmed/mobile/data/model/AuthResponse.kt
package com.appointmed.mobile.data.model

data class AuthResponse(
    val token: String? = null,
    val id: Long = 0,
    val name: String = "",
    val email: String = "",
    val role: String = "",
    val avatarUrl: String? = null,
    val avatarData: String? = null,
    val phone: String? = null,
    val address: String? = null,
    val gender: String? = null,
    val birthDate: String? = null,
    val bloodType: String? = null,
    val specialization: String? = null,
    val licenseNumber: String? = null,
    val clinicAddress: String? = null,
    val consultationFee: Double = 0.0
) {
    fun toUser(password: String? = null): User {
        return User(
            id = id,
            name = name,
            email = email,
            password = password ?: "",
            role = role,
            avatarUrl = avatarUrl,
            avatarData = avatarData,
            phone = phone,
            address = address,
            gender = gender,
            birthDate = birthDate,
            bloodType = bloodType,
            specialization = specialization,
            licenseNumber = licenseNumber,
            clinicAddress = clinicAddress,
            consultationFee = consultationFee
        )
    }
}
